nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -275,11 +280,31 @@ BtNavigator::navigateToPose()
       geometry_msgs::msg::PoseStamped goal_pose;
       blackboard_->get("goal", goal_pose);
 
-      feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-        feedback_msg->current_pose.pose, goal_pose.pose);
+      // Get current path points
+      nav_msgs::msg::Path current_path;
+      blackboard_->get("path", current_path);
+
+      // Calculate distance on the path
+      double distance_remaining =
+        nav2_util::geometry_utils::calculate_path_length(current_path);
+
+      if (distance_remaining <= transform_tolerance_) {
+        estimated_time_remaining_ = rclcpp::Duration::from_seconds(0.0);
+      } else {
+        // Get current speed
+        current_linear_speed_ = odom_smoother_->getTwist().linear.x;
```

Comment:
In case holonomic is used, you could to a `hypot(linear.x, linear.y)` so that you can capture the actual speed in all directions. This only really works for differential robots (non-ackermann, non-omnidirectional)

Reply:
@SteveMacenski how can I tell what type of robot is being used? It is true I assumed differential drive here.

Reply:
Actually, you can ignore that question, `hypot(linear.x, linear.y)` should work in all cases where the robot motion is in the xy plane.